phpBB/phpbb/feed/helper.php

```diff
@@ -105,15 +100,15 @@ public function generate_content($content, $uid, $bitfield, $options, $forum_id,
 		}
 
 		// Prepare some bbcodes for better parsing
-		$content	= preg_replace("#\[quote(=&quot;.*?&quot;)?:$uid\]\s*(.*?)\s*\[/quote:$uid\]#si", "[quote$1:$uid]<br />$2<br />[/quote:$uid]", $content);
+		$content = preg_replace("#\[quote(=&quot;.*?&quot;)?:$uid\]\s*(.*?)\s*\[/quote:$uid\]#si", "[quote$1:$uid]<br />$2<br />[/quote:$uid]", $content);
 
 		$content = generate_text_for_display($content, $uid, $bitfield, $options);
 
 		// Add newlines
 		$content = str_replace('<br />', '<br />' . "\n", $content);
 
 		// Convert smiley Relative paths to Absolute path, Windows style
-		$content = str_replace($this->phpbb_root_path . $this->config['smilies_path'], $this->get_board_url() . '/' . $this->config['smilies_path'], $content);
+		$content = str_replace($this->path_helper->get_web_root_path() . $this->config['smilies_path'], $this->get_board_url() . '/' . $this->config['smilies_path'], $content);
```

Comment:
I'm ~80% sure that you can change the path in the `text_formatter.renderer` service *before* calling `generate_text_for_display()` a few lines above, rather than replace it with a string function.

```php
$renderer->set_smilies_path($this->get_board_url() . '/' . $this->config['smilies_path']);
```

Reply:
You are technically right. It just wouldn't work for posts with the old bbcodes, would it?

I wish I could tell the `parse_attachments` function to work with the board url, then I wouldn't need the `path_helper` at all...

Reply:
That's correct, messages posted on 3.1 would have the relative path until the reparser catches up. Since the reparser works backwards starting from the newest posts, that should be a small window of time.

Reply:
I wonder if we can just assume that the reparser - more specifically cron - works as it should on every board out there or if there are misconfigurations that prevent it from working...

Otherwise I totally agree with you.

Reply:
If you think there's a chance the cron doesn't run you can leave the string replacement in and still set the correct path in the renderer. The other string replacements don't really do anything in 3.2 either. For instance, `"<br />"` are now `"<br>\n"` by default.